### PR TITLE
Edit groups through the UI

### LINKED ISSUE #10: Admin UI for group creation and editing
Background: the legacy react front-end (see background in #5) has a form for creating groups, but the shape of the data is different from what is needed for this project. I suggest taking a look at that code and seeing if it's worth migrating it over or writing this feature from scratch.

Done when: admins can create and edit groups using the UI

### PR DESCRIPTION
**Pull request by deammer**
Resolves #10

### Changes

- added the ability to edit existing groups
- changed the country selection to a dropdown

### Screenshot

Screen Shot 2021-03-15 at 3 54 47 PM

### CODE REVIEW

File: frontend/src/data/constants.ts [hunk 1]
```diff
@@ -275,3 +276,10 @@ export const COUNTRY_CODES_TO_NAME = {
   ZM: 'Zambia',
   ZW: 'Zimbabwe',
 }
+
+export const COUNTRY_CODE_OPTIONS = enumKeys(COUNTRY_CODES_TO_NAME).map(
+  (key) => ({
+    label: COUNTRY_CODES_TO_NAME[key],
+    value: key,
+  }),
```
Comment (bion) [inline frontend/src/data/constants.ts:283]: Suggestion (non-blocking): alternatively could do

```typescript
export const COUNTRY_CODE_OPTIONS = Object.entries(COUNTRY_CODES_TO_NAME).map(
  ([countryCode, countryName]) => ({
    label: countryName,
    value: countryCode
  })
)
```

File: frontend/src/pages/groups/GroupEditPage.tsx [hunk 2]
```diff
@@ -16,15 +20,17 @@ const GroupEditPage: FunctionComponent = () => {
...
 
   const onSubmit = (input: GroupUpdateInput) => {
-    // TODO support an updateGroup resolver on the backend
-    // updateGroup({ variables: { input } }).catch((error) => {
-    //   console.log(error)
-    // })
+    updateGroup({ variables: { id: parseInt(groupId, 10), input } }).catch(
+      (error) => {
+        console.error(error)
+      },
```
Comment (bion) [inline frontend/src/pages/groups/GroupEditPage.tsx]: Suggestion (non-blocking): no need for the wrapping lambda?

```typescript
.catch(console.error)
```

**Review by bion — APPROVED** [review]

File: frontend/src/data/constants.ts (hunk 1 above)
Comment (deammer) [reply frontend/src/data/constants.ts:283]: Ooooh I forgot about that method 🙏

File: frontend/src/pages/groups/GroupEditPage.tsx (hunk 2 above)
Comment (deammer) [reply frontend/src/pages/groups/GroupEditPage.tsx]: Oh yeah good... catch

**Pull request merged**